python/ray/train/tensorflow/tensorflow_predictor.py

```diff
@@ -59,25 +59,59 @@ def from_checkpoint(
             preprocessor=preprocessor,
         )
 
+    def tensorize(self, numpy_array: np.ndarray, dtype: tf.dtypes.DType) -> tf.Tensor:
+        tf_tensor = tf.convert_to_tensor(numpy_array, dtype=dtype)
+
+        # Off-the-shelf Keras Modules expect the input size to have at least 2
+        # dimensions (batch_size, feature_size). If the tensor for the column
+        # is flattened, then we unqueeze it to add an extra dimension.
+        if len(tf_tensor.shape) == 1:
+            tf_tensor = tf.expand_dims(tf_tensor, axis=1)
+        return tf_tensor
+
+    def untensorize(self, tensor: tf.Tensor) -> np.ndarray:
+        return tensor.numpy()
+
+    def model_predict(
+        self, tensor: Union[tf.Tensor, Dict[str, tf.Tensor]]
+    ) -> Union[tf.Tensor, Dict[str, tf.Tensor], List[tf.Tensor], Tuple[tf.Tensor]]:
+        # TensorFlow model objects cannot be pickled, therefore we use
+        # a callable that returns the model and initialize it here,
+        # instead of having an initialized model object as an attribute.
```

Comment:
How expensive is initializing the model? Could we lazily initialize a `self._model` variable that just gets removed on pickling (in `__getstate__`)? This way we avoid constructing the model multiple times if `model_predict` is called more than once

Reply:
Good point, updated to call this in init instead of in `predict`.

Predictors are not serializable in the first place, so we don't have to worry about it being stored in an attribute.